File: app/src/main/java/com/jajinba/pixabaydemo/contract/MainActivityContract.java
package com.jajinba.pixabaydemo.contract;


import androidx.annotation.NonNull;
import androidx.annotation.StringRes;

public interface MainActivityContract {

  interface View {
    void searchStart();

    void searchFinished(boolean isSuccess, @StringRes int errorMsg);
  }

  interface Presenter {
    void search(@NonNull String keyword);
  }

}

File: app/src/main/java/com/jajinba/pixabaydemo/contract/MainActivityContract.kt
package com.jajinba.pixabaydemo.contract

import androidx.annotation.StringRes

interface MainActivityContract {
  interface View {
    fun searchStart()
    fun searchFinished(isSuccess: Boolean, @StringRes errorMsg: Int)
  }

  interface Presenter {
    fun search(keyword: String)
  }
}
